src/main/java/Duke.java

```diff
@@ -1,10 +1,228 @@
+import java.util.ArrayList;
+import java.util.Scanner;
+
 public class Duke {
+    /* List of output messages shown to users */
+    // Greetings
+    private static final String MESSAGE_LOGO = " ____        _        \n"
+            + "|  _ \\ _   _| | _____ \n"
+            + "| | | | | | | |/ / _ \\\n"
+            + "| |_| | |_| |   <  __/\n"
+            + "|____/ \\__,_|_|\\_\\___|\n";
+    private static final String MESSAGE_INTRO = "Hello from\n" + MESSAGE_LOGO;
+    private static final String MESSAGE_BODY = "Hello! I'm Duke\n"
+            + "What can I do for you?";
+    private static final String MESSAGE_GOODBYE = "Bye. Hope to see you again soon!";
+    private static final String DIVIDER = "____________________________________________________________";
+
+    // Tasks related messages
+    private static final String MESSAGE_LIST_TASK = "Here are the tasks in your list:";
+    private static final String MESSAGE_DONE_TASK = "Nice! I've marked this task as done:";
+    private static final String MESSAGE_ADD_TASK = "Got it. I've added this task:";
+
+    /* List of user input commands */
+    private static final String COMMAND_TODO = "todo";
+    private static final String COMMAND_DEADLINE = "deadline";
+    private static final String COMMAND_EVENT = "event";
+    private static final String COMMAND_LIST = "list";
+    private static final String COMMAND_DONE = "done";
+    private static final String COMMAND_BYE = "bye";
+
+    /* List of user command parameters */
+    private static final String PARAM_AT = "/at";
+    private static final String PARAM_BY = "/by";
+
+    /**
+     * Handles user I/O.
+     */
+    private static final Scanner SC = new Scanner(System.in);
+
+    /**
+     * Stores a list of user tasks
+     */
+    private static final ArrayList<Task> TASKS = new ArrayList<Task>();
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        printWelcome();
+        while (SC.hasNextLine()) {
+            String input = getUserInput();
+            executeUserCommand(input);
+        }
```

Comment:
I like how your code is abstracted into functions, enhancing readability without need for comments. Good Job! 👍